Source/WebKit/Shared/Extensions/WebExtensionUtilities.mm

```diff
@@ -223,11 +223,15 @@ static bool validate(NSString *key, NSObject *value, id expectedValueType, NSStr
     return [NSString stringWithFormat:@"'%@', and '%@'", formattedInitialItems, list.lastObject];
 }
 
-bool validateDictionary(NSDictionary<NSString *, id> *dictionary, NSString *sourceKey, NSArray<NSString *> *requiredKeys, NSArray<NSString *> *optionalKeys, NSDictionary<NSString *, id> *keyTypes, NSString **outExceptionString)
+bool validateDictionary(NSDictionary<NSString *, id> *dictionary, NSString *sourceKey, NSArray<NSString *> *requiredKeys, NSDictionary<NSString *, id> *keyTypes, NSString **outExceptionString)
 {
+    ASSERT(keyTypes.count);
+
     NSMutableSet<NSString *> *remainingRequiredKeys = [[NSMutableSet alloc] initWithArray:requiredKeys];
     NSSet<NSString *> *requiredKeysSet = [[NSSet alloc] initWithArray:requiredKeys];
-    NSSet<NSString *> *optionalKeysSet = [[NSSet alloc] initWithArray:optionalKeys];
+    NSMutableSet<NSString *> *optionalKeysSet = [[NSMutableSet alloc] initWithArray:keyTypes.allKeys];
+
+    [optionalKeysSet minusSet:requiredKeysSet];
```

Comment:
Nit: I'd put these two lines together. Maybe a blank line before creating `optionalKeySet`? I just think it makes it more clear both are needed for the proper set up of that variable.